pkg/pillar/nireconciler/genericitems/dnsmasq.go

```diff
@@ -606,24 +626,57 @@ func (c *DnsmasqConfigurator) CreateDnsmasqConfig(buffer io.Writer, dnsmasq Dnsm
 		}
 	}
 
-	if router != "" {
+	// Prepare the set of all static routes to propagate to applications.
+	var staticRoutes []types.IPRoute
+	if !isIPv6 {
+		if dnsmasq.DHCPServer.AllOnesNetmask {
+			// Make the gateway reachable using a link-local route.
+			if gatewayIP != nil {
+				staticRoutes = append(staticRoutes, types.IPRoute{
+					DstNetwork: netutils.HostSubnet(gatewayIP),
+					Gateway:    net.IP{0, 0, 0, 0},
+				})
+			}
+			if subnet != nil && gatewayIP != nil {
+				staticRoutes = append(staticRoutes, types.IPRoute{
+					DstNetwork: subnet,
+					Gateway:    gatewayIP,
+				})
+			}
+		} else {
+			// With all-ones netmask disabled we use forwarding between apps
+			// on the same NI.
+			if subnet != nil {
+				staticRoutes = append(staticRoutes, types.IPRoute{
+					DstNetwork: subnet,
+					Gateway:    net.IP{0, 0, 0, 0},
```

Comment:
why the gateway is 0.0.0.0 here?

Reply:
This is a link-local route for the whole subnet, such as: `10.10.10.0/24 dev eth0 proto dhcp scope link`